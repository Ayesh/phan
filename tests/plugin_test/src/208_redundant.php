<?php
function check_assign_redundant(array $x) {
    if ($x[0] = 'a') {
        echo "should this be ==\n";
    }
    if (!$x[1] = false) {
        echo "should this be ==\n";
    }
}
