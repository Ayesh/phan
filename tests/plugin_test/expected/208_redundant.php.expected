src/208_redundant.php:2 PhanUnreferencedFunction Possibly zero references to function \check_assign_redundant()
src/208_redundant.php:3 PhanPluginNonBoolBranch Non bool value of type 'a' evaluated in if clause
src/208_redundant.php:3 PhanRedundantCondition Redundant attempt to cast $x[0] of type 'a' to truthy
src/208_redundant.php:6 PhanImpossibleCondition Impossible attempt to cast $x[1] of type false to truthy
src/208_redundant.php:6 PhanUnusedVariable Unused definition of variable $x
